### PR TITLE
Added OpenMP Library Package to Rumprun

### PR DESCRIPTION
**Pull request by VincentS**
Hey,

I packaged the OpenMP library for Rumprun to be able to run programs that use the OpenMP library.

Signed-off-by: Vincent Schwarzer vincent.schwarzer@yahoo.de

### CODE REVIEW

**Comment by anttikantee** [comment]
Hi, I can't see your actual change.

Also, can you submit your change to nginx as a separate pull request so that the nginx maintainer can decide what do with it.

**Comment by VincentS** [comment]
I just fixed my Pull Request snafu and will make a separate Pull Request for NGINX. 
You should see now my actual changes. Sorry for any inconvenience.

File: openmp/Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+all: bin/openmp
+
+bin/openmp:	build/openmp
+	mkdir -p include
+	cp gcc-build/omp.h include/
+	mkdir -p lib
+	cp gcc-build/.libs/libgomp.a lib/
+	rm -rf gcc-build
+	rm -rf gcc-source
```
Comment (anttikantee) [inline openmp/Makefile]: Why is the rm -rf performed here?  It's more polite to leave that choice up to the user to do via `make clean`

File: openmp/scripts/get_gcc_ver.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+. ../config.mk;
```
Comment (anttikantee) [inline openmp/scripts/get_gcc_ver.sh]: This is, strictly speaking, not legal.  The quoting of shell scripts and make is different.  If you need the value of CC, you need to use `make` to print it.

File: openmp/scripts/get_gcc_ver.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+gcc_ver=$($RUMPRUN_TOOLCHAIN_TUPLE-gcc -v 2>&1 >/dev/null|grep -Po  '(?<=gcc version\s)([0-9]\.[0-9])');
+gcc_ver_check=$(echo $gcc_ver | cut -d '.' -f 1)
+if [ $(( $gcc_ver_check )) -ge 5 ] && [ $(( $gcc_ver_check )) -lt 6 ]; then
```
Comment (anttikantee) [inline openmp/scripts/get_gcc_ver.sh]: _scratch_... why not just use `-eq 5`?

File: openmp/scripts/get_gcc_ver.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+. ../config.mk;
+gcc_ver=$($RUMPRUN_TOOLCHAIN_TUPLE-gcc -v 2>&1 >/dev/null|grep -Po  '(?<=gcc version\s)([0-9]\.[0-9])');
+gcc_ver_check=$(echo $gcc_ver | cut -d '.' -f 1)
+if [ $(( $gcc_ver_check )) -ge 5 ] && [ $(( $gcc_ver_check )) -lt 6 ]; then 
+	gcc_branch='gcc-5-branch';
+	echo $gcc_branch;
+	
+elif [ $(( $gcc_ver_check )) -ge 4 ] && [ $(( $gcc_ver_check )) -lt 5 ]; then
```
Comment (anttikantee) [inline openmp/scripts/get_gcc_ver.sh]: See above.

File: openmp/scripts/git_clone.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	exit 1;
+fi
+
+if [ -n $3 ]; then
+	git clone -b $3 $1 $2;
+	exit 0;
+else
+	git clone $1 $2;
+	exit 0;
+fi
```
Comment (anttikantee) [inline openmp/scripts/git_clone.sh]: Is this git clone script really necessary?  Isn't it just easier to do the same inline in the Makefile?

**Comment by anttikantee** [comment]
Thanks, I see it now.  I added a few comments.

Can you add more explicit instructions on how to test this system, or make a post about it on the mailing list?  I'd like to try it out and understand your work.  (it's not a requirement for the package to be accepted, but I'm personally curious)

**Comment by VincentS** [comment]
Hey,

I incorporated your feedback into the package and wrote a small tutorial for building and running an OpenMP application on Rumprun. I will write an short email over the mailing list today with a short overview about the new package.

**Comment by anttikantee** [comment]
Thanks. Can you include a more complex example on how you would you OpenMP in reality?  I assume it involves setting up some cluster, etc., so it would be nice to get it documented.

For example, can you run an OpenMP program on the host and delegate computation to Rumprun nodes?

(yea, it's better to get this discussion done on the mailing list, so looking forward to that)

**Comment by VincentS** [comment]
I think you are confusing OpenMP  with OpenMP<b>I</b> here. OpenMP is designed for multiprocessor/core shared memory systems were parallelism is achieved by using threads.
OpenMPI can be used for distributed as well as shared memory system.I am actually working on OpenMPI for Rumprun as well but it is not done yet.

There is an implementation of OpenMP called OpenMP\* by Intel which can be used for distributed computing but I didn't look into it yet.

Longer email regarding OpenMP and current state of my OpenMPI efforts on Rumprun over mailing list. ;)

**Pull request merged**